pkg/cache/redis.go

```diff
@@ -0,0 +1,115 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/tilezen/tapalcatl/pkg/state"
+	"github.com/tilezen/tapalcatl/pkg/tile"
+)
+
+type redisCache struct {
+	client *redis.Client
+}
+
+func (m *redisCache) Get(ctx context.Context, key string) ([]byte, error) {
+	bytes, err := m.client.Get(ctx, key).Bytes()
+	if err != nil {
+		if err == redis.Nil {
+			// Redis responds with a Nil error if there was a miss.
+			return nil, nil
+		}
+
+		return nil, err
+	}
+
+	return bytes, nil
+}
+
+func (m *redisCache) Set(ctx context.Context, key string, val []byte) error {
+	err := m.client.Set(ctx, key, val, 0).Err()
+	if err != nil {
+		return fmt.Errorf("error setting to redis: %w", err)
+	}
+
+	return nil
+}
+
+func (m *redisCache) GetTile(ctx context.Context, req *state.ParseResult) (*state.VectorTileResponseData, error) {
+	key := buildVectorTileKey(req)
+
+	item, err := m.Get(ctx, key)
+	if err != nil {
+		return nil, fmt.Errorf("error getting from redis: %w", err)
+	}
+
+	if item == nil {
+		return nil, nil
+	}
+
+	response, err := unmarshallVectorTileData(item)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+func (m *redisCache) SetTile(ctx context.Context, req *state.ParseResult, resp *state.VectorTileResponseData) error {
+	key := buildVectorTileKey(req)
+
+	marshalled, err := marshallVectorTileData(resp)
+	if err != nil {
+		return fmt.Errorf("error marshalling to redis: %w", err)
+	}
+
+	err = m.Set(ctx, key, marshalled)
```

Comment:
Do we want a TTL for the cache? we are still figuring out the storage structure, but it is likely that the server code is not aware of diff build release under the run_id (in your case the build_id)

Reply:
I can add in support for TTL, but I think any build system that adds support for incremental updates to a build_id will need to also push through cache invalidations.

Reply:
Added TTL in 4a83ddf.